src/git.md

```diff
@@ -0,0 +1,181 @@
+particular conflict. Edit the file to make the changes you want and remove the
+`<<<<<<<`, `=======` and `>>>>>>>` lines in the process. Second, check the
+surrounding code. If there was a conflict, its because someone else changed the
+same code you did. That means its likely there are some logical errors lying
+around too!
+
+Once you're all done fixing the conflicts, you need to stage the files that had
+conflicts in them via `git add`. Afterwards, run `git rebase --continue` to let
+git know that you've resolved the conflicts and it should finish the rebase.
+Finally, once the rebase has succeeded, you'll want to update the associated
+branch on your fork with `git push -f`.
+
+## Advanced Rebasing
+
+Sometimes, you may want to perform a more complicated rebase. There are two
+common scenarios that might call for this.
+
+If your branch contains multiple consecutive rewrites of the same code, or if
+the rebase conflicts are extremely severe, it is possible that just trying to
+reapply the changes you made on top of the updated code will be too much of a
+headache. In this case, you can use the interactive rebase feature via
+`git rebase -i master` to gain more control over the process. This allows you
+to choose to skip commits because they represent changes you no longer need,
+edit the commits that you do not skip, or change the order in which they are
+applied.
+
+The other common scenario is if you are asked to or want to "squash" multiple
+commits into each other. The most common example of this is likely if you
+forgot to run `x.py tidy` before committing. Your PR will need to be revised,
+but a single commit at the end with message "fixup tidy issue" is usually
+unhelpful, and it is easier for everyone else if you combine that commit with
+another that has a more meaningful commit message. In this case, you'll want to
+run `git rebase -i HEAD~2` to edit the last two commits and merge them
+together. Essentially, this reapplies the last two commits on top of your
+current branch; this is of course a no-op, since that is where they are
+already. However, by selecting the `-i` option, you give yourself the
+opportunity to edit the rebase first, just like before. This way you can
+request to have the most recent commit squashed into its parent.
+
+## No-Merge Policy
+
+The rust-lang/rust repo uses what is known as a "rebase workflow." This means
+that merge commits in PRs are not accepted. As a result, if you are running
+`git merge` locally, chances are good that you should be rebasing instead. Of
+course, this is not always true; if your merge will just be a fast-forward,
+like the merges that `git pull` usually performs, then no merge commit is
+created and you have nothing to worry about.
+
+There are a number of reasons for this decision and like all others, it is of
+course a tradeoff. The main advantage is the (mostly) linear commit history.
+This greatly simplifies bisecting. TODO: There are other advantages to a rebase
```

Comment:
Bisecting is the main one, yeah - rollups make this hard enough, it would be even more painful if the commit log didn't match the order PRs were merged.